### PR TITLE
Make compatible to fmt version 9.0.3

### PR DESCRIPTION
**Pull request by windrad6**

### CODE REVIEW

File: include/villas/nodes/websocket.hpp [hunk 1]
```diff
@@ -25,7 +26,7 @@ namespace villas {
 namespace node {
 
 /* Forward declarations */
-class NodeCompat;
+// class NodeCompat;
```
Comment (stv0g) [inline include/villas/nodes/websocket.hpp:29]: Please try to keep the forward declaration here as it reduces the amount of code which needs to get recompiled if `node_compat.hpp` changes.

The proper way of fixing this, to remove the code in the header which relies on the full declaration of NodeCompat to the .cpp file.

File: lib/path.cpp [hunk 2]
```diff
@@ -290,7 +290,7 @@ void Path::prepare(NodeList &nodes)
 
 		ret = pd->prepare(queuelen);
 		if (ret)
-			throw RuntimeError("Failed to prepare path destination {} of path {}", *pd->node, *this);
+			throw RuntimeError("Failed to prepare path destination {} of path {}", (pd->node)->getName(), this->toString());
```
Comment (stv0g) [inline lib/path.cpp:293]: Parentheses not required?

File: lib/path.cpp [hunk 3]
```diff
@@ -492,7 +492,7 @@ void Path::checkPrepared()
 			/* Check that all path sources provide a file descriptor for polling if fixed rate is disabled */
 			for (auto ps : sources) {
 				if (!(ps->getNode()->getFactory()->getFlags() & (int) NodeFactory::Flags::SUPPORTS_POLL))
-					throw RuntimeError("Node {} can not be used in polling mode with path {}", *ps->getNode(), *this);
+					throw RuntimeError("Node {} can not be used in polling mode with path {}", (ps->getNode())->getName(), this->toString());
```
Comment (stv0g) [inline lib/path.cpp:495]: Parentheses not required?

File: lib/path_source.cpp [hunk 4]
```diff
@@ -193,9 +193,9 @@ void MasterPathSource::writeToSecondaries(struct Sample *smps[], unsigned cnt)
 	for (auto sps : secondaries) {
 		int sent = sps->getNode()->write(smps, cnt);
 		if (sent < 0)
-			throw RuntimeError("Failed to write secondary path source {} of path {}", *sps->getNode(), *path);
+			throw RuntimeError("Failed to write secondary path source {} of path {}", (sps->getNode())->getName(), path->toString());
```
Comment (stv0g) [inline lib/path_source.cpp:196]: Parentheses not required?

File: lib/path_source.cpp [hunk 5]
```diff
@@ -193,9 +193,9 @@ void MasterPathSource::writeToSecondaries(struct Sample *smps[], unsigned cnt)
...
 		else if ((unsigned) sent < cnt)
-			path->logger->warn("Partial write to secondary path source {} of path {}", *sps->getNode(), *path);
+			path->logger->warn("Partial write to secondary path source {} of path {}", (sps->getNode())->getName(), path->toString());
```
Comment (stv0g) [inline lib/path_source.cpp:198]: Parentheses not required?

File: src/villas-pipe.cpp [hunk 6]
```diff
@@ -493,7 +493,7 @@ check:			if (optarg == endptr)
 
 		ret = node->getFactory()->stop();
 		if (ret)
-			throw RuntimeError("Failed to stop node type {}: reason={}", *node->getFactory(), ret);
+			throw RuntimeError("Failed to stop node type {}: reason={}", (node->getFactory())->getName(), ret);
```
Comment (stv0g) [inline src/villas-pipe.cpp:496]: Parentheses not required?

File: src/villas-signal.cpp [hunk 7]
```diff
@@ -255,15 +255,15 @@ check:			if (optarg == endptr)
 
 		ret = node->getFactory()->start(nullptr);
 		if (ret)
-			throw RuntimeError("Failed to intialize node type {}: reason={}", *node->getFactory(), ret);
+			throw RuntimeError("Failed to intialize node type {}: reason={}", (node->getFactory())->getName(), ret);
```
Comment (stv0g) [inline src/villas-signal.cpp:258]: Parentheses not required?

File: src/villas-signal.cpp [hunk 8]
```diff
@@ -304,7 +304,7 @@ out:			sample_decref(t);
 
 		ret = node->getFactory()->stop();
 		if (ret)
-			throw RuntimeError("Failed to de-intialize node type {}: reason={}", *node->getFactory(), ret);
+			throw RuntimeError("Failed to de-intialize node type {}: reason={}", (node->getFactory())->getName(), ret);
```
Comment (stv0g) [inline src/villas-signal.cpp:307]: Parentheses not required?

File: src/villas-test-rtt.cpp [hunk 9]
```diff
@@ -175,15 +175,15 @@ check:			if (optarg == endptr)
 
 		ret = node->getFactory()->start(&sn);
 		if (ret)
-			throw RuntimeError("Failed to start node-type {}: reason={}", *node->getFactory(), ret);
+			throw RuntimeError("Failed to start node-type {}: reason={}", (node->getFactory())->getName(), ret);
```
Comment (stv0g) [inline src/villas-test-rtt.cpp:178]: Parentheses not required?

File: src/villas-test-rtt.cpp [hunk 10]
```diff
@@ -225,11 +225,11 @@ check:			if (optarg == endptr)
 
 		ret = node->stop();
 		if (ret)
-			throw RuntimeError("Failed to stop node {}: reason={}", *node, ret);
+			throw RuntimeError("Failed to stop node {}: reason={}", node->getName(), ret);
 
 		ret = node->getFactory()->stop();
 		if (ret)
-			throw RuntimeError("Failed to stop node-type {}: reason={}", *node->getFactory(), ret);
+			throw RuntimeError("Failed to stop node-type {}: reason={}", (node->getFactory())->getName(), ret);
```
Comment (stv0g) [inline src/villas-test-rtt.cpp:232]: Parentheses not required?

File: include/villas/path.hpp [hunk 11]
```diff
@@ -126,6 +126,12 @@ class Path {
 		return os;
 	}
 
+	std::string toString() {
```
Comment (stv0g) [inline include/villas/path.hpp:129]: Open `{` in new line

**Review by stv0g — CHANGES_REQUESTED** [review]

File: src/villas-pipe.cpp [hunk 12]
```diff
@@ -452,7 +452,7 @@ check:			if (optarg == endptr)
 
 		ret = node->getFactory()->start(&sn);
 		if (ret)
-			throw RuntimeError("Failed to intialize node type {}: reason={}", *node->getFactory(), ret);
+			throw RuntimeError("Failed to intialize node type {}: reason={}", (node->getFactory())->getName(), ret);
```
[the remark this answers is not in the document]
Comment (windrad6) [reply src/villas-pipe.cpp:455]: This way it is more clear what happens and "better" readable. I would like to keep it if you don't insist on it

File: include/villas/nodes/websocket.hpp (hunk 1 above)
Comment (windrad6) [reply include/villas/nodes/websocket.hpp]: fixed

File: include/villas/nodes/websocket.hpp [hunk 13]
```diff
@@ -89,12 +90,18 @@ struct websocket_connection {
...
 		if (c.node)
-			os << ", node=" << *c.node;
+			os << ", node=" << c.node->getName();
 
 		os << ", mode=" << (c.mode == websocket_connection::Mode::CLIENT ? "client" : "server");
 
 		return os;
 	}
+
+	std::string toString() {
```
[the remark this answers is not in the document]
Comment (windrad6) [reply include/villas/nodes/websocket.hpp]: fixed

**Comment by stv0g** [comment]
The CI is not working again on the master branch 🥳 

Can you rebase the PR on the new master to see if the CI succeeds?

**Review by stv0g — APPROVED** [review]
I made final adjustments. Now we are ready to merge

**Pull request merged**